src/policies/BaseRewardDistributor.sol

```diff
@@ -0,0 +1,398 @@
+
+    // ========== MODIFIERS ========== //
+
+    modifier onlyAuthorized(bytes32 role_) {
+        ROLES.requireRole(role_, msg.sender);
+        _;
+    }
+
+    // ========== CONSTRUCTOR ========== //
+
+    /// @notice Constructor
+    ///
+    /// @param  kernel_             The Kernel address
+    /// @param  rewardTokenVault_   The ERC4626 vault token
+    /// @param  epochStartDate_     The timestamp when epoch 0 begins (midnight UTC of start date)
+    constructor(
+        address kernel_,
+        address rewardTokenVault_,
+        uint256 epochStartDate_
+    ) Policy(Kernel(kernel_)) {
+        if (rewardTokenVault_ == address(0)) revert RewardDistributor_InvalidAddress();
+        if (epochStartDate_ == 0) revert RewardDistributor_EpochIsZero();
+        _validateEpochStartOfDay(epochStartDate_);
+        REWARD_TOKEN = IERC20(IERC4626(rewardTokenVault_).asset());
+        REWARD_TOKEN_VAULT = IERC4626(rewardTokenVault_);
+        // Note: epochStartDate_ is truncated to uint40. Max uint40 is ~year 36812.
+        EPOCH_START_DATE = uint40(epochStartDate_);
+        // Disabled by default by PolicyEnabler
+    }
+
+    // ========== POLICY SETUP ========== //
+
+    /// @inheritdoc Policy
+    function configureDependencies() external override returns (Keycode[] memory dependencies) {
+        dependencies = new Keycode[](2);
+        dependencies[0] = toKeycode("ROLES");
+        dependencies[1] = toKeycode("TRSRY");
+
+        // ROLES is inherited from PolicyEnabler
+        ROLES = ROLESv1(getModuleAddress(dependencies[0]));
+        TRSRY = TRSRYv1(getModuleAddress(dependencies[1]));
+    }
+
+    /// @inheritdoc Policy
+    function requestPermissions()
+        external
+        view
+        virtual
+        override
+        returns (Permissions[] memory permissions)
+    {
+        Keycode trsryKeycode = toKeycode("TRSRY");
+
+        permissions = new Permissions[](1);
+        permissions[0] = Permissions({
+            keycode: trsryKeycode,
+            funcSelector: TRSRY.withdrawReserves.selector
+        });
+    }
+
+    function VERSION() external pure virtual returns (uint8 major, uint8 minor) {
+        major = 1;
+        minor = 0;
+        return (major, minor);
+    }
+
+    // ========== MERKLE ROOT MANAGEMENT ========== //
+
+    /// @notice Sets the Merkle root for an epoch
+    /// @dev    - Only callable by the ROLE_MERKLE_UPDATER role
+    ///         - Epoch start date must be at least 1 day after the last epoch start date
+    ///         - Epoch start date must be at the exact beginning of a day (midnight UTC)
+    ///         - Merkle tree cannot be updated once set
+    ///
+    /// @param  epochStartDate_ The epoch start date (timestamp) being set
+    /// @param  merkleRoot_     The Merkle root for the epoch's distribution
+    /// @return epochStartDate  The epoch start date that was set
+    function setMerkleRoot(
+        uint40 epochStartDate_,
```

Comment:
This is setting the merkle root for the period from `lastEpochStartDate` to `epochStartDate_ - 1`, correct?  The wording is a little confusing. Perhaps it should be epoch end date instead.

Reply:
The naming is still confusing.

How about:
- Rename this function to `endEpoch`
- Rename `epochStartDate_` to `epochEndDate_`

The naming throughout the contract should then reflect this. By calling end epoch, you're stating that the end of the epoch is `epochEndDate_` and the merkle root for the rewards is what's given. By implication, the next epoch starts 1 second after `epochEndDate_`.